sdk/src/main/scala/com/akkaserverless/javasdk/impl/valueentity/ValueEntitiesImpl.scala

```diff
@@ -164,11 +165,23 @@ final class ValueEntitiesImpl(_system: ActorSystem,
             context.deactivate() // Very important!
           }
 
+          val serializedPrimaryEffect = effect.primaryEffect match {
+            case UpdateState(state) => UpdateState(service.anySupport.encodeJava(state))
+            case DeleteState => DeleteState
+            case NoPrimaryEffect => NoPrimaryEffect
+          }
+          // FIXME something more that needs to be serialized?
+          val serializedSecondaryEffect = effect.secondaryEffect match {
+            case MessageReplyImpl(message, metadata, sideEffects) =>
+              MessageReplyImpl(service.anySupport.encodeJava(message), metadata, sideEffects)
+            case other => other
+          }
+
           val clientAction =
-            context.replyToClientAction(effect.secondaryEffect, allowNoReply = false, restartOnFailure = false)
+            context.replyToClientAction(serializedSecondaryEffect, allowNoReply = false, restartOnFailure = false)
 
-          if (!context.hasError && !effect.secondaryEffect.isInstanceOf[ErrorReplyImpl[_]]) {
-            val (nextState: Option[ScalaPbAny], action: Option[ValueEntityAction]) = effect.primaryEffect match {
+          if (!context.hasError && !serializedSecondaryEffect.isInstanceOf[ErrorReplyImpl[_]]) {
+            val (nextState, action) = serializedPrimaryEffect match {
```

Comment:
aren't those type annotations nice to have? I remember I was having trouble keeping track of where things were serialized and where they were not ;)

Reply:
I think I removed it because of some warnings. I will check and re-introduce if that's not the case.

Reply:
added back